datastore/query.js

```diff
@@ -0,0 +1,388 @@
+  var query = datastore.createQuery('Task')
+    .filter('priority =', 4)
+    .filter('done =', false)
+    .filter('created >', new Date('1990-01-01T00:00:00z'))
+    .filter('created <', new Date('2000-12-31T23:59:59z'));
+  // [END equal_and_inequality_range]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySort = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('priority')
+    .order('created');
+  // [END inequality_sort]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySortInvalidNotSame = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort_invalid_not_same]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('created');
+  // [END inequality_sort_invalid_not_same]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySortInvalidNotFirst = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort_invalid_not_first]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('created')
+    .order('priority');
+  // [END inequality_sort_invalid_not_first]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testLimit = function(callback) {
+  var datastore = this.datastore;
+
+  // [START limit]
+  var query = datastore.createQuery('Task')
+    .limit(5);
+  // [END limit]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testCursorPaging = function(callback) {
+  var pageSize = 1;
+
+  datastore.createQuery = this.datastore.createQuery;
+
+  // [START cursor_paging]
+  var query = datastore.createQuery('Task')
+    // Pagination is enabled by default in gcloud-node. To switch it off:
+    .autoPaginate(false)
+    .limit(pageSize);
+
+  function onQueryResults(err, results, nextQuery) {
+    if (err) {
+      // An error occurred while running the query.
+      return;
+    }
+
+    // If there are more results to retrieve, the cursor will have automatically
+    // been set on `nextQuery`.
+    if (nextQuery) {
+      datastore.runQuery(nextQuery, onQueryResults);
+    }
+  }
+
+  datastore.runQuery(query, onQueryResults);
+  // [END cursor_paging]
+
+  delete datastore.createQuery;
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testEventualConsistentQuery = function(callback) {
+  var query = this.ancestorQuery;
+
+  // jshint unused:false
+  // [START eventual_consistent_query]
+  datastore.runQuery(query, function(err, entities) {
+    if (!err) {
+      // Entities found.
+    }
+  });
+  // [END eventual_consistent_query]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testIndex1Query1 = function() {
+  // [START index1_query1]
+  // GQL is not supported in gcloud-node.
```

Comment:
Ah sorry, no need to add these (index*_query*). They are only going to show the GQL format string for them, we decided we don't need per-language snippets for them.